File: Lý Thuyết/6. Knowledge/Lesson_01/OOP/Demo/CodeDemoGeneric/Person.ts
export class Person<T, U> {

    private id: T;
	private name: U;
	
	constructor(id : T, name : U){
		this.id = id;
		this.name = name;
	}

	public getId(): T {
		return this.id;
	}

	public getName(): U {
		return this.name;
	}

	public setId(value: T) {
		this.id = value;
	}

	public setName(value: U) {
		this.name = value;
	}
}

File: Lý Thuyết/6. Knowledge/Lesson_06/Lifecycle/Component _ PureComponent/Shallow _ Deep Copy/VD4/entity/Person.java
package entity;

public class Person {

	private Address address;
	private Name name;

	public Person(Address address, Name name) {
		this.address = address;
		this.name = name;
	}

	public Person(Person otherPerson) {
		this.address = new Address(otherPerson.getAddress());
		this.name = new Name(otherPerson.getName());
	}

	public Address getAddress() {
		return address;
	}

	public Name getName() {
		return name;
	}

	@Override
	public String toString() {
		return name.toString() + address.toString();
	}
}
